src/integrations.ts

```diff
@@ -0,0 +1,40 @@
+import CleengService from '#src/services/integrations/cleeng/cleeng.service';
+import AccountService from '#src/services/account.service';
+import CleengAccountService from '#src/services/integrations/cleeng/cleeng.account.service';
+import CheckoutService from '#src/services/checkout.service';
+import CleengCheckoutService from '#src/services/integrations/cleeng/cleeng.checkout.service';
+import SubscriptionService from '#src/services/subscription.service';
+import CleengSubscriptionService from '#src/services/integrations/cleeng/cleeng.subscription.service';
+import InplayerAccountService from '#src/services/integrations/jwp/inplayer.account.service';
+import InplayerCheckoutService from '#src/services/integrations/jwp/inplayer.checkout.service';
+import SubscriptionJWService from '#src/services/integrations/jwp/inplayer.subscription.service';
+import CleengProfileService from '#src/services/integrations/cleeng/cleeng.profile.service';
+import InplayerProfileService from '#src/services/integrations/jwp/inplayer.profile.service';
+import ProfileService from '#src/services/profile.service';
+import { container, getModule } from '#src/container';
+import AppController from '#src/stores/AppController';
+
+const appController = getModule(AppController);
+
+appController.registerIntegration({
```

Comment:
tbh, I don't see how this is any better than having all the bindings in 1 place  with some if's. You've gotta look in 2 files (this one and container.ts) to see all the bindings and a third file to see where those bindings are actually bound. It's more files and lines of code to call the exact same bind methods.

Reply:
Do you suggest merging the container.ts and integrations.ts files? The only reason for separating the two is that the integrations logic depends on the AppController. However, it is still possible to resolve it from the container file...

Reply:
Yes, keeping all the bindings together in 1 file, but also, I think the registerIntegration logic is just adding unnecessary indirection. It would only make sense if we needed to dynamically create integrations at runtime, which we don't / can't because JS lacks any way to do the reflection or auto-discovery.

Reply:
I am okay with combining the files.

About the registration part, this helps publishers register custom identity providers that aren't productized without changing the implementation code, for example, for an Adobe TVE integration. You're right that this can't be used runtime, but it does help with the implementation when publishers have forked the repository and want to keep sync.

I'm open to better suggestions! 😄

Reply:
I think your latest proposal might already solve this, but just to finish the thread - I would be OK with integrations having individual files where they register themselves (so the main app references that 1 file where the integration binds its own modules.) I don't think we need to add an extra layer of decoupling with the `appController.registerIntegration` logic.

Reply:
Yes, I agree! I did split the container initialization from the registration/bindings to fix some import issues we will be having later. The core only cares about the container instance and `getModule` helper but shouldn't care how the dependencies are bound.